### PR TITLE
Add mini simplex application tests

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Comment by peterrum** [comment]
I have added alternative versions of the Trilinos/PETSc tests, which do not use these libraries.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -43,14 +43,14 @@ jobs:
         sudo apt-get install -y software-properties-common
         sudo add-apt-repository ppa:ginggs/deal.ii-9.2.0-backports
         sudo apt-get update
-        sudo apt-get install -yq --no-install-recommends numdiff  libp4est-dev trilinos-all-dev petsc-dev
+        sudo apt-get install -yq --no-install-recommends numdiff  libp4est-dev trilinos-all-dev petsc-dev libmetis-dev
```
Comment (peterrum) [inline .github/workflows/linux.yml:46]: Moved to #10789.

File: tests/simplex/poisson_02.cc [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
+  error_table.set_scientific("error", true);
+
+  error_table.write_text(deallog.get_file_stream());
+  deallog << std::endl;
+}
+
+int
+main()
+{
+  initlog();
```
Comment (kronbichler) [inline tests/simplex/poisson_02.cc:630]: :+1:

**Comment by peterrum** [comment]
I have made the changes.

**Pull request merged**